test/extended/oauth/client_secret.go

```diff
@@ -0,0 +1,96 @@
+)
+
+var _ = g.Describe("[sig-auth][Feature:OAuthServer]", func() {
+	defer g.GinkgoRecover()
+	oc := exutil.NewCLI("oauthclient-secret-with-plus")
+	ctx := context.Background()
+
+	g.Describe("ClientSecretWithPlus", func() {
+		g.It(fmt.Sprintf("should create oauthclient"), func() {
+			g.By("create oauth client")
+			oauthClient, err := oc.AdminOauthClient().OauthV1().OAuthClients().Create(ctx, &oauthv1.OAuthClient{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "oauth-client-with-plus",
+				},
+				Secret:            "secret+with+plus",
+				RedirectURIs:      []string{"https://www.google.com"},
+				GrantMethod:       oauthv1.GrantHandlerAuto,
+				ScopeRestrictions: []oauthv1.ScopeRestriction{{ExactValues: []string{"user:full"}}},
+			}, metav1.CreateOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+			oc.AddResourceToDelete(oauthv1.GroupVersion.WithResource("oauthclients"), oauthClient)
+
+			g.By("create synthetic identity, user, and binding")
+
+			user, err := oc.AdminUserClient().UserV1().Users().Create(ctx, &userv1.User{
+				TypeMeta: metav1.TypeMeta{},
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "user-for-plus",
+				},
+				FullName: "fake user",
+			}, metav1.CreateOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+			oc.AddResourceToDelete(userv1.GroupVersion.WithResource("users"), user)
+
+			oauthAuthorizeToken, err := oc.AdminOauthClient().OauthV1().OAuthAuthorizeTokens().Create(ctx, &oauthv1.OAuthAuthorizeToken{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "oauth-client-with-plus-with-more-than-thirty-two-characters-in-this-very-long-name",
+				},
+				ClientName:          oauthClient.Name,
+				ExpiresIn:           100000000,
+				RedirectURI:         "https://www.google.com",
+				Scopes:              []string{"user:full"},
+				UserName:            user.Name,
+				UserUID:             string(user.UID),
+				CodeChallenge:       "",
+				CodeChallengeMethod: "",
+			}, metav1.CreateOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+			oc.AddResourceToDelete(oauthv1.GroupVersion.WithResource("oauthauthorizetokens"), oauthAuthorizeToken)
+
+			g.By("querying for a token")
+			oauthRoute, err := oc.AdminRouteClient().RouteV1().Routes("openshift-authentication").Get(ctx, "oauth-openshift", metav1.GetOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			tokenPost, err := http.NewRequest("POST", "http://"+oauthRoute.Status.Ingress[0].Host+"/oauth/token?"+
+				"grant_type=authorization_code&"+
+				"code=oauth-client-with-plus-with-more-than-thirty-two-characters-in-this-very-long-name&"+
+				"client_id=oauth-client-with-plus&"+
+				"client_secret=secret%2Bwith%2Bplus",
+				nil)
+			o.Expect(err).NotTo(o.HaveOccurred())
+			requestDump, err := httputil.DumpRequest(tokenPost, true)
+			o.Expect(err).NotTo(o.HaveOccurred())
+			framework.Logf("%v", string(requestDump))
+
+			// we don't really care if this URL is safe
+			tr := &http.Transport{
+				TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
```

Comment:
you'll want to add `Proxy: ProxyFromEnvironment,` to the transport so that the test does not break in proxied environment